CoreScriptsRoot/CoreScripts/PurchasePromptScript2.lua

```diff
@@ -399,6 +462,58 @@ PurchaseDialog.Parent = RobloxGui
 			xOffset = xOffset + 32
 		end
 
+--Gui State Control
+local function setPurchaseDialogOpen(isOpen)
+	if isOpen then
```

Comment:
This function seems like a complicated way of saying `PurchaseDialog.Visible = isOpen`